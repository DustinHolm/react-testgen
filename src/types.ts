export interface ImportBlock {
    sourceFile: string,
    isInternal: boolean,
    imports: Import[]
}

export interface Import {
    isDefault: boolean,
    name: string,
    element?: ExportableElement
}

export interface Export {
    name: string,
    isDefault: boolean,
    element?: ExportableElement
}

export enum ExportType {
    Class,
    Function,
    Interface,
    Variable
}

export interface ExportableElement {
    type: ExportType,
    returnsJSX: boolean
}

export interface ClassElement extends ExportableElement {
    methods: MethodElement[],
    constructors: Constructor[]
}

export interface MethodElement extends FunctionElement {
    name: string
}

export interface FunctionElement extends ExportableElement {
    returnType: string,
    parameters: Attribute[]
}

export interface InterfaceElement extends ExportableElement {
    attributes: Attribute[]
}

export interface VariableElement extends ExportableElement {
    varType: string,
    varValue: string
}

export interface Attribute {
    name: string,
    type: string
}

export interface Constructor {
    parameters: Attribute[]
}