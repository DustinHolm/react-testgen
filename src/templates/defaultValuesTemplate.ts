import { FunctionElement } from "../types"
import { getReturnForType } from "./common"

export const createJSXDefaults = (name: string): string => {
    return `\
    ${name}Return = <span data-testid={"${name}"} />
    ${name} = jest.fn(() => ${name}Return)\
`
}

export const createFunctionDefaults = (name: string, element: FunctionElement): string => {
    const returnValue = getReturnForType(element.returnType)

    return `\
    ${name}Return = ${returnValue}
    ${name} = jest.fn(() => ${name}Return)\
`
}
