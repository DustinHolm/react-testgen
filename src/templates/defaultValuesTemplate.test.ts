import { ExportType, FunctionElement } from "../types"
import { createFunctionDefaults, createJSXDefaults } from "./defaultValuesTemplate"

describe("defaultValuesTemplate Tests", () => {
    describe("createJSXDefaults Tests", () => {
        test("With name", () => {
            const returnedString = createJSXDefaults("test")
            expect(returnedString).toMatchSnapshot()
        })
    })

    describe("createFunctionDefaults Tests", () => {
        test("With name", () => {
            const element: FunctionElement = {
                type: ExportType.Function,
                returnsJSX: false,
                returnType: "testType",
                parameters: []
            }
            const returnedString = createFunctionDefaults("test", element)
            expect(returnedString).toMatchSnapshot()
        })
    })
})