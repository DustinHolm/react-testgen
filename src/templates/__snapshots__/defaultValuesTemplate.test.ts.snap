// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultValuesTemplate Tests createFunctionDefaults Tests With name 1`] = `
"    testReturn = undefined //TODO: Set this manually
    test = jest.fn(() => testReturn)"
`;

exports[`defaultValuesTemplate Tests createJSXDefaults Tests With name 1`] = `
"    testReturn = <span data-testid={\\"test\\"} />
    test = jest.fn(() => testReturn)"
`;
